packages/trader/src/Modules/Trading/Components/Form/ContractType/contract-type-display.tsx

```diff
@@ -14,6 +14,8 @@ type TDisplay = {
 };
 
 const Display = ({ is_open, name, list, onClick, value }: TDisplay) => {
+    React.useMemo(() => (window as any).hj('event', `selected_${value}_contract_type`), [value]);
```

Comment:
only want to execute once after we change contract_type. Without memo ContractType.Display rerenders and sends event after ContractType.Dialog closes when clicking outside or clicking on the same element to close the dialog.

Reply:
@henry-deriv it's better to do it in contract-type-widget. @akmal-deriv added rudderstack event for trade type selection before (`action: 'choose_trade_type'`), it'd be better if you could place this in there too.

Reply:
@henry-deriv it's better to do it in contract-type-widget. @akmal-deriv added rudderstack event for trade type selection before (`action: 'choose_trade_type'`), it'd be better if you could place this in there too.